libpromises/verify_vars.c

```diff
@@ -212,6 +212,11 @@ PromiseResult VerifyVarPromise(EvalContext *ctx, const Promise *pp,
             if (res.status == FNCALL_FAILURE)
             {
                 /* We do not assign variables to failed fn calls */
+                if (EvalContextGetPass(ctx) == 3) {
+                    // If we still fail at pass3, report an error
+                    Log(LOG_LEVEL_ERR, "While setting variable '%s' in bundle '%s', function '%s' failed - skipping",
+                                       pp->promiser, PromiseGetBundle(pp)->name, fp->name);
```

Comment:
Please change the log level to `LOG_LEVEL_VERBOSE`. This information is useful, but we don't want to introduce too much log spam in the default case.

Reply:
> @nickanderson what do you think?

In principal, I think having the ability to easily find these is great. The only thing that I didn't like skimming over it was a hard coded 3 which i think is for the 3 passes of evaluation. There are tickets to make evaluation details configurable, so I think it would be best to not hardcode 3. Also, I agree with LOG_LEVEL_VERBOSE, we don't need a log explosion from everyones existing policy.

Reply:
I agree.